src/context/ColonyContext.tsx

```diff
@@ -46,9 +52,19 @@ export const ColonyContextProvider = ({
   children: ReactNode;
 }) => {
   const { colonyName } = useParams<{ colonyName: string }>();
+  const [prevColonyName, setPrevColonyName] = useState<string>();
+
   const { state: locationState } = useLocation();
   const [isPolling, setIsPolling] = useState(!!colonyName);
 
+  /* Update polling state when routing between colonies */
```

Comment:
This ensures we don't inadvertently restart polling when switching between colonies.